Merge Review composer UX - Meeds-io/MIPs#54

MayTekayaa opened the pull request:
This change will merge the review composer UX MIP.

MayTekayaa committed:
fix: Improve the post CTA from a space - MEED-2070  - meeds-io/MIPs#54 (#2707)

This change will improve the post CTA from a space.

MayTekayaa committed:
feat: Adjust the style of the activity stream title when the activity composer is enabled for users - MEED-2402 - Meeds-io/MIPs#54  (#2756)

This change will adjust the style of the activity stream title when the activity composer is enabled for users.

MayTekayaa committed:
fix: Remove placeholder when editing an activity - MEED-2444 - Meeds-io/MIPs#54  (#2794)

This change will remove placeholder when editing an activity.

MayTekayaa committed:
fix: Improve the toolbar post CTA - MEED-2401 - Meeds-io/MIPs#54 (#2799)

This chane will improve the toolbar post CTA.

MayTekayaa committed:
feat: Add the ability to send a kudos from the composer - MEED-2071 - Meeds-io/MIPs#54  (#2811)

This change will add the ability to send a kudos from the composer .

MayTekayaa committed:
feat: Add CTAs below the post CTA - MEED-2072 - Meeds-io/MIPs#54  (#2813)

This fix will add  send kudos and create poll CTAs below the post CTA.

MayTekayaa committed:
feat: Disable send kudos button when no receiver has been added - MEED-2418 - Meeds-io/MIPs#54 (#2814)

This change will fix the fetch space API console error.

MayTekayaa committed:
feat: Adjust the CTAs display from the general stream - MEED-2073 - Meeds-io/MIPs#54  (#2833)

This change will adjust the CTAs display from the general stream.

MayTekayaa committed:
feat: Review the activity stream filter option - MEED-2422 - Meeds-io/MIPs#54 (#2855)

This change will review the activity stream filter option.

MayTekayaa committed:
fix: Adjust the placeholder depending the audience - MEED-2464 - Meeds-io/MIPs#54  (#2858)

This change will adjust the placeholder depending the audience.

MayTekayaa committed:
feat: Add a property to post to your network or not - MEED-2075 - Meeds-io/MIPs#54  (#2865)

This change will add a property to post to your network or not.

MayTekayaa committed:
feat: Add the ability to choose the audience from composer - MEED-2462  - Meeds-io/MIPs#54 (#2881)

This change will add the ability to choose the audience from composer.

MayTekayaa committed:
fix: Change the composer filter texts - MEED-2503 - Meeds-io/MIPs#54 (#2884)

This change will change the composer filter texts.

MayTekayaa committed:
fix: Enable the main stream composer property - MEED-2548 - Meeds-io/MIP#54  (#2910)

This change will enable the main stream composer property.

MayTekayaa committed:
fix: Hide the reset filter button when using mobile - MEED-2543 - Meeds-io/MIP#54  (#2911)

This change will hide the reset filter button when using mobile.

MayTekayaa committed:
fix: Mention a user when changing the audience - MEED-2463 - Meeds-io/MIP#54  (#2908)

This change will mention a user according to the selected audience.

MayTekayaa committed:
feat: Add the ability to send a kudos from genral stream- MEED-2076 - Meeds-io/MIP#54  (#2917)

This change will add the ability to send a kudos from genral stream.

MayTekayaa committed:
fix: Adjust the kudos CTA from general stream - MEED-2568 - Meeds-io/MIP#54 (#2932)

This change will adjust the kudos CTA from general stream.

MayTekayaa committed:
fix: Display the audience choice when only create an activity - MEED-2572 - Meeds-io/MIP#54 (#2937)

This change will display the audience choice when only create an activity.

MayTekayaa committed:
fix: Add some enhancements to the the activity toolbar icons - MEED-2581 - Meeds-io/MIP#54  (#2939)

This change will add some enhancements to the the toolbar activity icons.

 #fix: Add some enhancements to the the activity toolbar icons - MEED-2581 - Meeds-io/MIP#54  (#2944)

MayTekayaa committed:
fix: Review the display of the composer drawer when space audience chosen - MEED-2569 - Meeds-io/MIP#54 (#2945)

This change will review the display of the composer drawer when space audience chosen.

MayTekayaa committed:
fix: Fix the post attach image in general stream - MEED-2587 -Meeds-io/MIP#54 (#2952)

This change will fix the post attach image in general stream.

MayTekayaa committed:
MEED-2075 Fix the post button activation when the property is set to false

rdenarie commented on webapp/portlet/src/main/webapp/vue-apps/activity-stream/components/toolbar/ActivityStreamFilterDrawer.vue:
```diff
@@ -0,0 +1,128 @@
+<!--
+  This file is part of the Meeds project (https://meeds.io/).
+  Copyright (C) 2022 Meeds Association
```
```suggestion
  Copyright (C) 2023 Meeds Association
```

MayTekayaa replied:
done!

MayTekayaa committed:
fix: Fix the post button activation when the property is set to false - MEED-2608 -Meeds-io/MIP#54  (#2968)

This change will fix the post button activation when the property is set to false.

rdenarie commented on webapp/portlet/src/main/webapp/groovy/social/webui/UISocialPortalApplicationHead.gtmpl:
```diff
@@ -65,4 +65,5 @@
 
      eXo.env.portal.SpaceHomeLayoutResetEnabled = <%=featureService.isFeatureActiveForUser("SpaceHomeLayoutReset", userName)%>;
      eXo.env.portal.PublisherRolePromotionFeatureEnabled = <%=featureService.isFeatureActiveForUser("publisherRolePromotion", userName)%>;
+     eXo.env.portal.postToNetworkEnabled = <%=PropertyManager.getProperty("exo.feature.PostToNetwork.enabled")%>;
```
@boubaker should we use featureService instead of propertyManager ?
```suggestion
     eXo.env.portal.postToNetworkEnabled =  <%=featureService.isFeatureActiveForUser("PostToNetwork", userName)%>;
```

boubaker replied:
Yes, good idea, if we need to enable the feature for some users only.

MayTekayaa committed:
Use feature service instead of property manager for postToNetwork property
